### PR TITLE
[SQL] Add some missing DataFrame functions.

### PR DESCRIPTION
**Pull request by marmbrus**
- as with a `Symbol`
- distinct
- sqlContext.emptyDataFrame
- move add/remove col out of RDDApi section

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 1]
```diff
@@ -214,7 +214,12 @@ trait DataFrame extends RDDApi[Row] {
   /**
    * Returns a new [[DataFrame]] with an alias set.
    */
-  def as(name: String): DataFrame
+  def as(alias: String): DataFrame
+
+  /**
+   * Returns a new [[DataFrame]] with an alias set.
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: Can you add (Scala-specific) to the front? i.e.

```scala
(Scala-specific) Returns a new [[DataFrame]] with an alias set.
```

**Comment by rxin** [comment]
LGTM

**Comment by marmbrus** [comment]
test this please

**Comment by marmbrus** [comment]
test this please

**Pull request closed without merging**